bundles/org.openhab.ui/web/src/components/widgets/system/oh-video-webrtc.vue

```diff
@@ -0,0 +1,112 @@
+<template>
+  <video
+    ref="videoPlayer"
+    :autoplay="this.startManually ? false : true"
+    :controls="this.hideControls ? false : true"
+    playsinline
+    style="max-width: 100%; max-height: 100%;">
+    Sorry, your browser doesn't support embedded videos.
+  </video>
+</template>
+
+<script>
+export default {
+  name: 'OhVideoWebRTC',
+  props: {
+    src: { type: String },
+    config: { type: Object },
+    startManually: { type: Boolean },
+    hideControls: { type: Boolean }
+  },
+  data () {
+    return {
+      webrtc: null,
+      isClosed: false // if we closed the webrtc stream
+    }
+  },
+  watch: {
+    src (value) {
+      this.startPlay()
+    }
+  },
+  mounted () {
+    if (this.src) {
+      this.startPlay()
+    }
+  },
+  beforeDestroy () {
+    this.closeConnection()
+  },
+  methods: {
+    closeConnection () {
+      this.isClosed = true
+      if (this.webrtc) {
+        this.webrtc.close()
+      }
+    },
+    startPlay () {
+      this.closeConnection()
+      this.isClosed = false
+      const self = this
+      if (!self.src) {
+        return
+      }
+      self.webrtc = new RTCPeerConnection({
+        iceServers: [
+          {
+            urls: ['stun:stun.l.google.com:19302']
```

Comment:
Thanks.
I'm not familiar with WebRTC so I'm just getting learned on that!

Some good articles:
https://www.html5rocks.com/en/tutorials/webrtc/infrastructure/
https://medium.com/tenable-techblog/using-webrtc-ice-servers-for-port-scanning-in-chrome-ce17b19dd474

This is arguably a good hardcoded default, but would you say it's right not be made overridable?

Reply:
>I'm not familiar with WebRTC so I'm just getting learned on that!
Its a neat technology but complicated technology ! I had to do a lot with it in a former life. 

Yeah, so 99% of the time everyone uses the google stun servers, although i have run my own coturn server (https://github.com/coturn/coturn) in a commercial setting years ago when needing to support private TURN connections (which is rare these days).  I figured there was little chance to need to change this , but can add this as a "advanced option" for sure if we think it's important!

Reply:
> Its a neat and complicated technology ! I had to do a lot with it in a former life.

Sure looks like it - I had my share of complicated technologies in former lives but not this one 😄 

> I figured there was little chance to need to change this , but can add this as a "advanced option" for sure if we think it's important!

You're the subject matter expert so if you don't believe it's necessary, then fine! I only wanted to challenge you with it because I suspected users who are adamant about a wholly locally-hosted solution might not appreciate the hardcoded Google server.